### PR TITLE
Feature/apl 540 node

### PR DESCRIPTION
**Pull request by oleksiylukin**
This work is not complete but contains a lot of changes to P2P code and preliminary implementation of P2P file downloading. Work will be continued in other branch after merge to implemet-sharding

### CODE REVIEW

File: apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/peer/statcheck/FileDownloadDecision.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+ * Copyright © 2018-2019 Apollo Foundation
+ */
+package com.apollocurrency.aplwallet.apl.core.peer.statcheck;
+
+/**
+ *
+ * @author al
+ */
+public enum FileDownloadDecision {
+    AbsOK, //network is 100% consistent
```
Comment (AndrewBoyarsky) [inline apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/peer/statcheck/FileDownloadDecision.java]: Refactor states to upper case

File: apl-utils/src/main/java/com/apollocurrency/aplwallet/apl/util/ChunkedFileOps.java [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+        CheckSum cs = new CheckSum();
+        cs.update(data);
+        lastWRChunkCrc=cs.finish();
+        if(lastWRChunkCrc!=crc){
+            throw new BadCheckSumException(absPath.toString());
+        }
+        if(!absPath.toFile().exists()){
+            Files.createFile(absPath);
+        }
+        RandomAccessFile rf = new RandomAccessFile(absPath.toFile(),"rw");
```
Comment (AndrewBoyarsky) [inline apl-utils/src/main/java/com/apollocurrency/aplwallet/apl/util/ChunkedFileOps.java]: Cache opened file or wrap into try-with-resources

File: apl-utils/src/main/java/com/apollocurrency/aplwallet/apl/util/ChunkedFileOps.java [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+        return res;
+    }
+    
+    public int readChunk(int offset, int size, byte[] dataBuf) throws IOException{
+        int res;
+        if(!absPath.toFile().exists()){
+           res=-2;
+           return res;
+        }        
+        RandomAccessFile rf = new RandomAccessFile(absPath.toFile(),"r");
```
Comment (AndrewBoyarsky) [inline apl-utils/src/main/java/com/apollocurrency/aplwallet/apl/util/ChunkedFileOps.java]: Cache opened file or wrap into try-with-resources

File: apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/peer/PeerLoaderThread.java [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+ *
+ * @author alukin@gmail.com
+ */
+class PeerLoaderThread implements Runnable {
+    private static final Logger LOG = LoggerFactory.getLogger(PeerLoaderThread.class);
+    private final List<String> defaultPeers;
+    private EpochTime timeService;
+    private final List<Future<String>> unresolvedPeers;
+
+    public PeerLoaderThread(List<String> defaultPeers, List<Future<String>> unresolvedPeers, EpochTime timeService) {
```
Comment (yuriy-larin-firstbridge) [inline apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/peer/PeerLoaderThread.java]: NULL checks are highly desirable.

File: apl-core/src/test/java/com/apollocurrency/aplwallet/apl/testutil/EntityProducer.java [hunk 5]
```diff
@@ -7,6 +7,7 @@
 package com.apollocurrency.aplwallet.apl.testutil;
```
Comment (yuriy-larin-firstbridge) [inline apl-core/src/test/java/com/apollocurrency/aplwallet/apl/testutil/EntityProducer.java]: Apollo copyright is missing

**Pull request merged**